tests/functional/DatadirTest.php

```diff
@@ -0,0 +1,437 @@
+                    ],
+                ],
+            ],
+        ];
+        // phpcs:enable
+
+        $this->runAppWithConfig($config);
+
+        $manifestFilePath = $this->temp->getTmpFolder() . '/out/tables/out.c-my.testmetadata.manifest';
+        $manifestData = json_decode((string) file_get_contents($manifestFilePath), true);
+        $this->assertArrayHasKey('destination', $manifestData);
+        $this->assertArrayHasKey('metadata', $manifestData);
+        $this->assertArrayHasKey('column_metadata', $manifestData);
+
+        $expectedTableMetadata = [
+            [
+                'key' => 'KBC.name',
+                'value' => 'testmetadata',
+            ],
+        ];
+
+        $expectedColumnMetadata = [
+            'ID' => [
+                [
+                    'key' => 'KBC.datatype.type',
+                    'value' => 'NUMBER',
+                ],
+                [
+                    'key' => 'KBC.datatype.nullable',
+                    'value' => true,
+                ],
+                [
+                    'key' => 'KBC.datatype.basetype',
+                    'value' => 'NUMERIC',
+                ],
+                [
+                    'key' => 'KBC.datatype.length',
+                    'value' => '38,0',
+                ],
+                [
+                    'key' => 'KBC.type',
+                    'value' => 'NUMBER',
+                ],
+            ],
+            'NAME' => [
+                [
+                    'key' => 'KBC.datatype.type',
+                    'value' => 'TEXT',
+                ],
+                [
+                    'key' => 'KBC.datatype.nullable',
+                    'value' => true,
+                ],
+                [
+                    'key' => 'KBC.datatype.basetype',
+                    'value' => 'STRING',
+                ],
+                [
+                    'key' => 'KBC.datatype.length',
+                    'value' => '200',
+                ],
+                [
+                    'key' => 'KBC.type',
+                    'value' => 'TEXT',
+                ],
+            ],
+            'NOTNULL' => [
+                [
+                    'key' => 'KBC.datatype.type',
+                    'value' => 'TEXT',
+                ],
+                [
+                    'key' => 'KBC.datatype.nullable',
+                    'value' => false,
+                ],
+                [
+                    'key' => 'KBC.datatype.basetype',
+                    'value' => 'STRING',
+                ],
+                [
+                    'key' => 'KBC.datatype.length',
+                    'value' => '200',
+                ],
+                [
+                    'key' => 'KBC.type',
+                    'value' => 'TEXT',
+                ],
+            ],
+        ];
+
+        $expectedColumns = [
+            'ID', 'NAME', 'NOTNULL',
+        ];
+
+        $this->assertEquals($expectedTableMetadata, $manifestData['metadata']);
```

Comment:
btw, static methods should not be called statically, ale to je https://www.youtube.com/watch?v=YpMTWjwnGxU :)